### PR TITLE
feature/assignment15

### PR DESCRIPTION
**Pull request by dheerajsrinath**
Create one class having 1 method with different parameters like fullName(String firstName),
fullName(String firstName, String lastName), fullName(String firstName, String lastName, String
middleName), call all this method method and print in the console.
Note - Method overloading concept

### CODE REVIEW

File: src/main/java/Assignment15.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class Name{
+
```
Comment (PranavHanwante) [inline src/main/java/Assignment15.java]: remove intent

File: src/main/java/Assignment15.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    static String fullName(String firstName)
+    {
+        return firstName;
+    }
+    static String fullName(String firstName, String lastName)
```
Comment (PranavHanwante) [inline src/main/java/Assignment15.java]: intent

File: src/main/java/Assignment15.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        return firstName +" "+ lastName;
+    }
+    static String fullName(String firstName, String lastName, String middleName)
```
Comment (PranavHanwante) [inline src/main/java/Assignment15.java]: intent

File: src/main/java/Assignment15.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        return firstName + " "+ lastName+" "+ middleName;
+    }
+
```
Comment (PranavHanwante) [inline src/main/java/Assignment15.java]: remove intent

File: src/main/java/Assignment15.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+}
+public class Assignment15 {
```
Comment (PranavHanwante) [inline src/main/java/Assignment15.java:17]: intent

File: src/main/java/Assignment15.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    public static void main(String[] args){
+        Name methodOverload = new Name();
+        String x = methodOverload.fullName("Barack");
+        String y = methodOverload.fullName("Barack","Obama");
+        String z = methodOverload.fullName("Barack","Obama","Hussein");
+        System.out.println("Method 1 "+x);
+        System.out.println("Method 2 "+y);
+        System.out.println("Method 3 "+z);
+
+
```
Comment (PranavHanwante) [inline src/main/java/Assignment15.java]: remove intent

File: src/main/java/Assignment15.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+}
+public class Assignment15 {
+    public static void main(String[] args){
+        Name methodOverload = new Name();
+        String x = methodOverload.fullName("Barack");
+        String y = methodOverload.fullName("Barack","Obama");
+        String z = methodOverload.fullName("Barack","Obama","Hussein");
+        System.out.println("Method 1 "+x);
+        System.out.println("Method 2 "+y);
+        System.out.println("Method 3 "+z);
```
Comment (PranavHanwante) [inline src/main/java/Assignment15.java]: shortcut
  System.out.println("Method 1 "+  methodOverload.fullName("Barack"));
  do for remaining also

File: src/main/java/Assignment15.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
+class Name{
```
Comment (PranavHanwante) [inline src/main/java/Assignment15.java]: Give a meaning full name for class

File: src/main/java/Assignment15.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
+class FullNameOverload{
+static String fullName(String firstName)
+{
+    return firstName;
+}
+static String fullName(String firstName, String lastName)
```
Comment (PranavHanwante) [inline src/main/java/Assignment15.java:6]: intent

File: src/main/java/Assignment15.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+{
+    return firstName +" "+ lastName;
+}
+static String fullName(String firstName, String lastName, String middleName)
```
Comment (PranavHanwante) [inline src/main/java/Assignment15.java:10]: intent

File: src/main/java/Assignment15.java (hunk 9 above)
Comment (PranavHanwante) [inline src/main/java/Assignment15.java]: Intent

File: src/main/java/Assignment15.java (hunk 10 above)
Comment (PranavHanwante) [inline src/main/java/Assignment15.java]: Intent

File: src/main/java/Assignment15.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+{
+return firstName + " "+ lastName+" "+ middleName;
+}
+}
+public class Assignment15 {
```
Comment (PranavHanwante) [inline src/main/java/Assignment15.java]: Intent

File: src/main/java/com/assignment/Assignment15.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
+package com.assignment;
+
+public class Assignment15 {
+    public String fullName(String firstName) {
```
Comment (PranavHanwante) [inline src/main/java/com/assignment/Assignment15.java]: intent

File: src/test/java/com/assignmentTest/Assignment15Test.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Assignment15Test {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Test
+    public void testGetFullName() {
+
```
Comment (PranavHanwante) [inline src/test/java/com/assignmentTest/Assignment15Test.java]: remove intent

File: src/test/java/com/assignmentTest/Assignment15Test.java [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+        Assignment15 assignment15 = new Assignment15();
+        logger.info(assignment15.fullName("Barrack"));
+
+        Assertions.assertEquals(assignment15.fullName("Barrack"),"Barrack","Not full name");
+        logger.info(assignment15.fullName("Barrack", "Obama"));
+        Assertions.assertEquals(assignment15.fullName("Barrack", "Obama"),"Barrack Obama", "Not full name");
```
Comment (PranavHanwante) [inline src/test/java/com/assignmentTest/Assignment15Test.java]: put the remaining message in a new line = , "Not full name");

File: src/test/java/com/assignmentTest/Assignment15Test.java [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+        Assignment15 assignment15 = new Assignment15();
+        logger.info(assignment15.fullName("Barrack"));
+
+        Assertions.assertEquals(assignment15.fullName("Barrack"),"Barrack",errorMessage);
+        logger.info(assignment15.fullName("Barrack", "Obama"));
+        Assertions.assertEquals(assignment15.fullName("Barrack", "Obama"),"Barrack Obama", errorMessage);
+
+        logger.info(assignment15.fullName("Barrack", "Hussain", "Obama"));
+        Assertions.assertEquals(assignment15.fullName("Barrack", "Hussain", "Obama"),
+                "Barrack Hussain Obama",errorMessage);
```
Comment (PranavHanwante) [inline src/test/java/com/assignmentTest/Assignment15Test.java]: Barrack is repeating more than one time so put it in one string, check for others

**Review by akhil7000 — APPROVED** [review]

**Pull request merged**